docs/lib/DEBUGGING.md

```diff
@@ -60,22 +68,46 @@ License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 (gdb) ...
 ```
 
-The following is the minimum set of commands you need to send to the device to flash and run the binary (replace '2331' by '3333' if you use pyOCD or OpenOCD as a debug server):
-
-```bash
-(gdb) target remote localhost:2331
-(gdb) monitor endian little
-(gdb) monitor reset
-(gdb) monitor halt
-(gdb) monitor semihosting enable
-(gdb) monitor speed 1000
-(gdb) monitor loadbin BUILD/${target}/${your_app}.bin 0
-(gdb) monitor flash device = ${device_name}
-(gdb) load BUILD/${target}/${your_app}.elf
-(gdb) file BUILD/${target}/${your_app}.elf
-(gdb) call uvisor_api.debug_semihosting_enable()
-```
-the call to 'uvisor_api.debug_semihosting_enable' is required to enable semihosting debug printing.
+The following is the minimum set of commands you need to send to the device to flash and run the binary:
+
+* For J-Link debugger:
+
+  ```bash
+  (gdb) target remote localhost:2331
+  (gdb) monitor endian little
+  (gdb) monitor reset
+  (gdb) monitor halt
+  (gdb) monitor semihosting enable
+  (gdb) monitor speed 1000
+  (gdb) monitor loadbin BUILD/${target}/${your_app}.bin 0
+  (gdb) monitor flash device = ${device_name}
+  (gdb) load BUILD/${target}/${your_app}.elf
+  (gdb) file BUILD/${target}/${your_app}.elf
+  (gdb) call uvisor_api.debug_semihosting_enable()
+  ```
+
+* For PyOCD debugger:
+
+  ```bash
+  (gdb) target remote localhost:3333
+  (gdb) monitor reset
+  (gdb) monitor halt
+  (gdb) load BUILD/${target}/${your_app}.elf
+  (gdb) file BUILD/${target}/${your_app}.elf
+  (gdb) call uvisor_api.debug_semihosting_enable()
+  ```
+
+---
+**IMPORTANT**
+* The call to `uvisor_api.debug_semihosting_enable` is required to enable semihosting debug printing.
+  The reason for this "safety latch" is that the device faults on semihosting prints if a debugger is not connected.
+  On some devices it's hard to tell programmatically if a debugger is connected so semihosting is disabled by default and is enabled using this "safety latch".
+
+* If you wish to debug uVisor core as well you have to add the uVisor symbols:
```

Comment:
Please change to "If you also wish to debug the uVisor core, you must add the uVisor symbols:"

Reply:
OK